File: src - java/Esercizi/Occorrenze/Occorrenze.java
	//Metodo privato
	private int _occorrenze(Nodo n, int elem) {
		if (n == null)
			return 0;

		int occ = 0; // accumulatore occorrenze

		if (n.elem == elem)
			occ++;

		if (elem > n.elem) // il segno > deve essere coerente con
		                   // la convenzione stabilita nel metodo
		                   // aggiungiElem()
			occ += _occorrenze(n.des, elem);
		else
			occ += _occorrenze(n.sin, elem);

		return occ;
	}

	//Metodo pubblico
	public int occorrenze(int elem) {
		return _occorrenze(root, elem);
	}

File: src - c++/Esercizi/Occorrenze/Occorrenze.cpp
//Metodo privato

unsigned int AlberoBinario::_Occorrenze(const PNodo& n,
                                       const TElem& el) const {
  if (!n) //Se l'albero con radice in n è vuoto...
    return 0; //...il numero di occorrenze è pari a zero.
    
  int occ = 0;
  
  if (n->el == el)
    occ++;
  
  if (el > n->el) //il segno > deve essere coerente con la convenzione
                  //stabilita per l'inserimento degli elementi nell'albero
    occ = occ + _Occorrenze(n->des, el);
  else  
    occ = occ + _Occorrenze(n->sin, el);
  
  return occ;
}

//Metodo pubblico

unsigned int AlberoBinario::Occorrenze(const TElem& el) const {
  return _Occorrenze(root, el);  
}
